File: data/transcoder_evaluation_gfg/go/TRIANGULAR_NUMBERS_1.go
package main

import (
		"fmt"
	"math"
	"os"
	"unsafe"
)

func min(x int, y int) int {
	if x < y {
		return x
	}
	return y
}
func max(x int, y int) int {
	if x > y {
		return x
	}
	return y
}
func cmpfunc(a unsafe.Pointer, b unsafe.Pointer) int {
	return *(*int)(a) - *(*int)(b)
}


func f_gold(num int) bool {
	if num < 0 {
		return false
	}
	var c int = (num * (-2))
	var b int = 1
	var a int = 1
	var d int = (b * b) - a*4*c
	if d < 0 {
		return false
	}
	var root1 float32 = float32((float64(-b) + math.Sqrt(float64(d))) / float64(a*2))
	var root2 float32 = float32((float64(-b) - math.Sqrt(float64(d))) / float64(a*2))
	if root1 > 0 && math.Floor(float64(root1)) == float64(root1) {
		return true
	}
	if root2 > 0 && math.Floor(float64(root2)) == float64(root2) {
		return true
	}
	return false
}
//TOFILL
func main() {
	var (
		n_success int     = 0
		param0    []int = []int{1, 3, 6, 10, 55, 48, 63, 72, 16, 85}
	)
	for i := int(0); i < len(param0[:]); i++ {
		if f_filled(param0[i]) == f_gold(param0[i]) {
			n_success += 1
		}
	}
	fmt.Print("#Results:", " ", n_success, ", ", len(param0[:]))
	os.Exit(0)
}

File: data/transcoder_evaluation_gfg/c/TRIANGULAR_NUMBERS_1.c
// Copyright (c) 2019-present, Facebook, Inc.
// All rights reserved.
//
// This source code is licensed under the license found in the
// LICENSE file in the root directory of this source tree.
//


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>

int min(int x, int y) { return (x < y)? x: y; }
int max(int x, int y) { return (x > y)? x: y; }
int cmpfunc (const void * a, const void * b) {return ( *(int*)a - *(int*)b );}
int len (int arr [ ]) {return ((int) (sizeof (arr) / sizeof (arr)[0]));}
void sort (int arr [ ], int n) {qsort (arr, n, sizeof(int), cmpfunc);}

bool f_gold ( int num ) {
  if ( num < 0 ) return false;
  int c = ( - 2 * num );
  int b = 1, a = 1;
  int d = ( b * b ) - ( 4 * a * c );
  if ( d < 0 ) return false;
  float root1 = ( - b + sqrt ( d ) ) / ( 2 * a );
  float root2 = ( - b - sqrt ( d ) ) / ( 2 * a );
  if ( root1 > 0 && floor ( root1 ) == root1 ) return true;
  if ( root2 > 0 && floor ( root2 ) == root2 ) return true;
  return false;
}


bool f_filled ( int num ) {}

int main(void) {
    int n_success = 0;
    int param0[] = {1,3,6,10,55,48,63,72,16,85};
    for(int i = 0; i < len(param0); ++i)
    {
        if(f_filled(param0[i]) == f_gold(param0[i]))
        {
            n_success+=1;
        }
    }
    printf("#Results:", " ", n_success, ", ", len(param0));
    return 0;
}